Src/Fido2/AttestationFormat/AppleAppAttest.cs

```diff
@@ -0,0 +1,124 @@
+                // Allow expired leaf cert in development environment
+                chain.ChainPolicy.VerificationTime = credCert.NotBefore.AddSeconds(1);
+            }
+
+            if (!chain.Build(credCert))
+            {
+                throw new Fido2VerificationException("Failed to build chain in Apple AppAttest attestation: " + chain.ChainStatus.FirstOrDefault().StatusInformation.ToString());
+            }
+
+            // 2. Create clientDataHash as the SHA256 hash of the one-time challenge your server sends to your app before performing the attestation, and append that hash to the end of the authenticator data (authData from the decoded object).
+            // 3. Generate a new SHA256 hash of the composite item to create nonce.
+            // 4. Obtain the value of the credCert extension with OID 1.2.840.113635.100.8.2, which is a DER - encoded ASN.1 sequence.Decode the sequence and extract the single octet string that it contains. Verify that the string equals nonce.
+            // Steps 2 - 4 done in the "apple" format verifier
+            Apple apple = new();
+            (var attType, var trustPath) = apple.Verify(attStmt, authenticatorData, clientDataHash);
+
+            // 5. Create the SHA256 hash of the public key in credCert, and verify that it matches the key identifier from your app.
+            Span<byte> credCertPKHash = stackalloc byte[32];
+            SHA256.HashData(credCert.GetPublicKey(), credCertPKHash);
+            var keyIdentifier = Convert.FromHexString(credCert.GetNameInfo(X509NameType.SimpleName, false));
+            if (!credCertPKHash.SequenceEqual(keyIdentifier))
+            {
+                throw new Fido2VerificationException("Public key hash does not match key identifier in Apple AppAttest attestation");
+            }
+
+            // 6. Compute the SHA256 hash of your app's App ID, and verify that it’s the same as the authenticator data's RP ID hash.
+            var appId = GetAppleAppIdFromCredCertExtValue(credCert.Extensions);
+            Span<byte> appIdHash = stackalloc byte[32];
+            SHA256.HashData(appId, appIdHash);
+            if (!appIdHash.SequenceEqual(AuthData.RpIdHash))
+            {
+                throw new Fido2VerificationException("App ID hash does not match RP ID hash in Apple AppAttest attestation");
+            }
+
+            // 7. Verify that the authenticator data's counter field equals 0.
+            if (AuthData.SignCount != 0)
+            {
+                throw new Fido2VerificationException("Sign count does not equal 0 in Apple AppAttest attestation");
+            }
+
+            // 8. Verify that the authenticator data's aaguid field is either appattestdevelop if operating in the development environment, or appattest followed by seven 0x00 bytes if operating in the production environment.
+            if (!AuthData.AttestedCredentialData.AaGuid.Equals(devAaguid) && !AuthData.AttestedCredentialData.AaGuid.Equals(prodAaguid))
```

Comment:
Maybe the apple env should be a config that we validate?
ie. expecting from dev-> verify Aaguid matches dev

Reply:
I can see where that could be useful if you wanted to not accept dev devices in a production only environment.